### ISSUE TITLE
Concatenate two fields when extracting the batch iterator

### ISSUE THREAD

**Issue by joanPlepi** [pytorch/text#1090]
## ❓ Questions and Help

**Description**
I have two input text fields, and one label. If I use the default batch iterators, I will get for both input text fields a batch of input sequences, that are padded. 
However, I want to concatenate the both input text fields, and after apply the corresponding padding to create the batch. At the moment I am just using the collate_fn of pytorch DataLoader, but I was wondering if there is any way to concatenate those input text fields in torchtext.
I would expect to be something similar to collocate_fn, but I would not have to implement the padding part.

**Comment by zhangguanheng66** [comment]
I think it's good that you are switch to DataLoader. We will remove the `Field` class soon.
